### PR TITLE
Windows binary install instructions, small binary install corrections…

### PR DESCRIPTION
**Pull request by jchidley**
I had a number of issues getting rust-analyzer to work on my windows machine. Mostly down to my poor understanding of how this worked. I have adjusted to documentation to make it clearer, I hope, for other people

I also discovered that the "install it from source" instructions needed a `cargo build` before issuing the `cargo xtask` command.

There are conflicts between the proxy installation of `rustup` and the `binary installed` version (owing to the proxy of rustup). I have added some notes to make it clear that this should not be mixed and matched.

For the record, I use this with helix. I like it a lot. (I have used VS Code and neovim editors recently).

### CODE REVIEW

**Comment by jchidley** [comment]
I accidentally added a second ` git checkout release` to the build from source instructions

File: manual.adoc [hunk 1]
```diff
@@ -149,12 +151,23 @@ Make sure that `~/.local/bin` is listed in the `$PATH` variable and use the appr
...
+. Open a new `Command Prompt` or `PowerShell` to use `rust-analyzer` 
+
 Alternatively, you can install it from source using the command below.
 You'll need the latest stable version of the Rust toolchain.
 
 [source,bash]
 ----
 $ git clone https://github.com/rust-lang/rust-analyzer.git && cd rust-analyzer
+$ git checkout release
+$ cargo build --release
```
Comment (lnicola) [inline manual.adoc:169]: No, you don't need to do this.

Comment (jchidley) [reply manual.adoc:169]: By default, it's using the nightly install, hence the checkout release. `cargo xtask` doesn't run without a `cargo build` first, on my system

Comment (jchidley) [reply manual.adoc:169]: It's working now but there have been a lot of state changes on my end, including upgrading `rustup` and fiddling with rust-analyzer's config. Before it was consistently refusing to recognise the "xtask" part of the cargo command, hence my fallback to `cargo build` first. That was a frustrating waste of time.

I'm not going to do any further diagnosis it works on my system now with the latest rust, rustup and rust-analyzer.

File: manual.adoc [hunk 2]
```diff
@@ -149,12 +151,21 @@ Make sure that `~/.local/bin` is listed in the `$PATH` variable and use the appr
 
 You don't have to use `~/.local/bin`, any other path like `~/.cargo/bin` or `/usr/local/bin` will work just as well.
 
+On Windows:
+
+. Download a suitable Windows specific release e.g. `rust-analyzer-x86_64-pc-windows-msvc.zip`
+. Extract the files
+. Copy the `rust-analyzer.exe` file to a suitable directory (e.g. `C:\tools`)
+. Add that directory to the user's *path* _environmental variable_
```
Comment (bjorn3) [inline manual.adoc:159]:
```suggestion
. Add that directory to the user's *PATH* _environmental variable_
```

File: manual.adoc [hunk 3]
```diff
@@ -149,12 +151,21 @@ Make sure that `~/.local/bin` is listed in the `$PATH` variable and use the appr
 
 You don't have to use `~/.local/bin`, any other path like `~/.cargo/bin` or `/usr/local/bin` will work just as well.
 
+On Windows:
+
+. Download a suitable Windows specific release e.g. `rust-analyzer-x86_64-pc-windows-msvc.zip`
+. Extract the files
+. Copy the `rust-analyzer.exe` file to a suitable directory (e.g. `C:\tools`)
+. Add that directory to the user's *PATH* _environment variable_
+. Open a new `Command Prompt` or `PowerShell` to use `rust-analyzer`
```
Comment (Veykril) [inline manual.adoc:160]: There is no need to invoke the binary manually. That's your LSP capable text editors job. Running the binary on its own doesn't do anything.

Comment (jchidley) [reply manual.adoc:160]: Obviously I wasn't clear here. What I meant was restart the shell to get rust-analyzer to work. Can you suggest alternative wording?

**Pull request closed without merging**